### PR TITLE
Separate out sections of ExchangeModal into hooks

### PR DESCRIPTION
**Pull request by jinchung**
This pulls out different related pieces of logic into separate hooks.

This also fixes:
- Swap to deposit output conversion error for assets that are non-18-decimal (e.g, WBTC)
- Checking allowance was comparing raw vs non-raw data
- Was not passing along underlying price when underlying does not exist in wallet for deposits (causing NaNs on transaction list)
- Withdraw max only worked for assets where the user had the underlying asset
- If an input currency in swap was not selected, a user would still be able to type into the native input (and this would cause a crash)
- Removes redundant or unused code
- SendSheet to reuse the max input balance logic
- Swap currency reserve subscription would not trigger for when the selected default input currency of a deposit would require a swap (e.g, if depositing WBTC but a user didn't have WBTC and would therefore require an ETH->WBTC swap)
- Disable the max button when the input currency is not selected

Tested with the following scenarios:
* eth to token swap
* token to eth swap
* token to token swap
* deposit DAI with DAI
* deposit to cX with no underlying X in wallet
* deposit DAI with MKR as input
* withdraw

### CODE REVIEW

File: src/utils/__tests__/contract.test.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { greaterThan } from '../../helpers/utilities';
 import Contract from '../contract';
 
-const { getAllowance } = Contract;
+const { getRawAllowance } = Contract;
```
Comment (brunobar79) [inline src/utils/__tests__/contract.test.js:4]: should we start running this on CI?

File: src/redux/explorer.js [hunk 2]
```diff
@@ -206,6 +207,7 @@ const listenOnAddressMessages = socket => dispatch => {
   });
 
   socket.on(messages.ADDRESS_TRANSACTIONS.APPENDED, message => {
+    logger.log('txns appended', get(message, 'payload.transactions', []));
```
[the remark this answers is not in the document]
Comment (jinchung) [reply src/redux/explorer.js:210]: Correct - that would be the "txns received" logic. This is only for when txns have been appended (new txns).

File: src/utils/__tests__/contract.test.js (hunk 1 above)
Comment (jinchung) [reply src/utils/__tests__/contract.test.js:4]: that would be a good idea - I think the other tests we will still need to update and fix first!

File: src/helpers/utilities.js [hunk 3]
```diff
@@ -215,10 +217,12 @@ export const multiply = (numberOne, numberTwo) =>
  * @param  {Number}   numberTwo
  * @return {String}
  */
-export const divide = (numberOne, numberTwo) =>
-  BigNumber(`${numberOne}`)
+export const divide = (numberOne, numberTwo) => {
+  if (!(numberOne || numberTwo)) return 0;
```
[the remark this answers is not in the document]
Comment (jinchung) [reply src/helpers/utilities.js]: good point!

**Pull request merged**